packages/api/src/controllers/access-control.test.ts

```diff
@@ -0,0 +1,175 @@
+      publicPlaybackId = publicStream.playbackId;
+      let id = uuid();
+      gatedAsset = await db.asset.create({
+        id,
+        name: "test-storage",
+        createdAt: Date.now(),
+        playbackId: await generateUniquePlaybackId(id),
+        playbackPolicy: {
+          type: "signed",
+        },
+        status: {
+          phase: "ready",
+          updatedAt: Date.now(),
+        },
+        userId: nonAdminUser.id,
+      });
+    });
+
+    it("should create a gate stream and allow playback with given public key", async () => {
+      client.jwtAuth = adminToken;
+      const res2 = await client.post("/access-control/gate", {
+        stream: `video+${gatedPlaybackId}`,
+        pub: signingKey.publicKey,
+        type: "jwt",
+      });
+
+      expect(res2.status).toBe(204);
+    });
+
+    it("should not allow playback on not existing streams or public keys", async () => {
+      const res = await client.post("/access-control/gate", {
+        stream: `video+0000000000`,
+        pub: signingKey.publicKey,
+        type: "jwt",
+      });
+      expect(res.status).toBe(403);
+      const res2 = await client.post("/access-control/gate", {
+        stream: `video+${gatedPlaybackId}`,
+        pub: "00000000000000",
+        type: "jwt",
+      });
+      expect(res2.status).toBe(403);
+    });
+
+    it("should not allow playback if stream is gated an pub is missing", async () => {
+      const res2 = await client.post("/access-control/gate", {
+        stream: `video+${gatedPlaybackId}`,
+        type: "jwt",
+      });
+      expect(res2.status).toBe(403);
+    });
+
+    it("should not allow playback if stream and public key does not share the same owner", async () => {
+      const otherStream = await client.post("/stream", {
+        name: "test",
+        playbackPolicy: {
+          type: "signed",
+        },
+      });
+      const otherStreamPlaybackId = (await otherStream.json()).playbackId;
+      const res = await client.post("/access-control/gate", {
+        stream: `video+${otherStreamPlaybackId}`,
+        pub: signingKey.publicKey,
+        type: "jwt",
+      });
+      expect(res.status).toBe(403);
+    });
+
+    it("should allow playback on public playbackId", async () => {
+      client.jwtAuth = adminToken;
+      const res = await client.post("/access-control/gate", {
+        stream: `video+${publicPlaybackId}`,
+        type: "jwt",
+      });
+      expect(res.status).toBe(204);
+    });
+
+    it("should allow playback if playbackPolicy is not specified", async () => {
+      client.jwtAuth = nonAdminToken;
+      const stream = await client.post("/stream", {
+        name: "test",
+      });
+      client.jwtAuth = adminToken;
+      const streamPlaybackId = (await stream.json()).playbackId;
+      const res = await client.post("/access-control/gate", {
+        stream: `video+${streamPlaybackId}`,
+        type: "jwt",
+        pub: signingKey.publicKey,
```

Comment:
Is this required on this case? Might wanna ensure it works if it is not specified as well, like the previous test

Reply:
WDYM? This is creating a stream without a `playbackPolicy` specified and assuring that `204` is returned

Reply:
But it's calling the `/gate` API with a public key. I mean that the playback should work even without any public key on the payload, instead of specifying a random one (e.g. there could be a bug in the API where it requires **some** JWT in the `/gate` API, even if it doesn't validate anything about it)

Reply:
Oh I get it, yeah, there is a specific test for that `"should allow playback on public playbackId"` which has no public key specified. I will add another point to that where both the publicKey is missing in the payload and the playbackPolicy in the content